pdb2pqr/biomolecule.py

```diff
@@ -501,19 +501,27 @@ def assign_termini(self, chain, neutraln=False, neutralc=False):
         if len(chain.residues) == 0:
             text = f'Error: chain "{chain.chain_id}" has 0 residues!'
             raise IndexError(text)
-        # Set the N-Terminus/ 5' Terminus
+
         res0 = chain.residues[0]
+        reslast = chain.residues[-1]
+        # Check if chain is cyclic. Amide distance ranges between 1.325 - 1.346
+        if util.distance(res0.map["N"].coords, reslast.map["C"].coords) < 1.35:
+            # If the chain is cyclic, don't apply termini.
+            return
+
+        # Set the N-Terminus/ 5' Terminus
         if isinstance(res0, aa.Amino):
             res0.is_n_term = True
-            if isinstance(res0, aa.PRO) or neutraln:
+            # If N is bonded to more than one heavy atom switch to neutral-nterm
+            heavy_bonds = [a for a in res0.map["N"].bonds if a.name[0] != "H"]
+            if neutraln or len(heavy_bonds) > 1:
```

Comment:
This is the generalization I memtioned to cover non-standard residues with N bonded to > 2 heavy atoms instead of just PRO